src/dilpack/clustering/spectral.py

```diff
@@ -0,0 +1,91 @@
+import numpy as np
+from sklearn.neighbors import radius_neighbors_graph
+
+
+class SpectralClustering:
+    def __init__(self, X, num_clusters):
+        self.K = num_clusters
+        self.max_iterations = 1000
+        self.num_examples = X.shape[0]
+        self.num_features = X.shape[0]
+
+    # Get eigenvectors and eigenvalues
+    def get_eigs_from_laplacian(self, X):
+
+        # Compute graph Laplacian from data
+        W = radius_neighbors_graph(X, 0.4, mode="distance", include_self=True).toarray()
```

Comment:
Not really sure how strict Prof is with this. Another approach is to pass a lamda method in __init__ that can be called to preprocess X using that passed method. That way you can reference radius_neighbors_graph in your notebook instead of here. 

Though for the python package this dependency should not be an issue.

Reply:
Thanks for this comment, I think I can revisit on another PR if I decide to implement it this way. For now though I think this will do.